xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -448,18 +444,30 @@ struct AppModel: ModelProtocol {
         environment: AppEnvironment,
         gatewayURL: String
     ) -> Update<AppModel> {
-        // If URL given is not valid, reset back to original value.
-        guard isValidWebURL(gatewayURL) else {
-            var model = state
-            model.gatewayURLTextField = model.gatewayURL
-            model.isGatewayURLTextFieldValid = true
-            return Update(state: model)
+        var fallback = state
+        fallback.gatewayURLTextField = state.gatewayURL
+        fallback.isGatewayURLTextFieldValid = true
+
+        // If URL given is not valid, fall back to original value.
+        guard let url = URL(string: gatewayURL) else {
+            return Update(state: fallback)
+        }
+
+        // If URL given is not HTTP, fall back to original value.
+        guard url.isHTTP() else {
+            return Update(state: fallback)
         }
+
         var model = state
         model.gatewayURL = gatewayURL
         model.gatewayURLTextField = gatewayURL
         model.isGatewayURLTextFieldValid = true
+
+        // Persist to UserDefaults
         AppDefaults.standard.gatewayURL = gatewayURL
```

Comment:
Similar to my clipboard comment the other day, I find myself wondering if `AppDefaults` should also be available via the `AppEnvironment`

Reply:
Fair point. Xcode does expose UserDefaults for unit testing, so we don't need to mock. However, there is a good argument to be made to make this part of environment anyway.

AppDefaults is used in several places. This change would touch a number of other files, so I suggest following up in another PR.

Reply:
https://github.com/subconsciousnetwork/subconscious/issues/413